bluemira/equilibria/run.py

```diff
@@ -274,24 +436,27 @@ def _get_sof_eof_opt_problems(self, psi_sof, psi_eof):
                 current_constraints += deepcopy(self._coil_cons)
 
             eq_constraints = deepcopy(self.eq_constraints)
-            for con in eq_constraints:
-                if isinstance(con, (PsiBoundaryConstraint, PsiConstraint)):
-                    con.target_value = psi_boundary / (2 * np.pi)
-            for con in current_constraints:
-                if isinstance(con, (PsiBoundaryConstraint, PsiConstraint)):
-                    con.target_value = psi_boundary / (2 * np.pi)
-
-            problem = self._make_opt_problem(
-                eq, optimiser, max_currents, current_constraints, eq_constraints
+            for constraints in (eq_constraints, current_constraints):
+                for con in constraints:
+                    if isinstance(con, (PsiBoundaryConstraint, PsiConstraint)):
+                        con.target_value = psi_boundary / (2 * np.pi)
+
+            opt_problems.append(
+                self._make_opt_problem(
+                    eq, optimiser, max_currents, current_constraints, eq_constraints
+                )
             )
 
-            opt_problems.append(problem)
-
         return opt_problems
 
     def _make_opt_problem(
-        self, eq, optimiser, max_currents, current_constraints, eq_constraints
-    ):
+        self,
+        eq: Equilibrium,
+        optimiser: Optimiser,
+        max_currents: np.ndarray,
+        current_constraints: List[OptimisationConstraint],
+        eq_constraints: MagneticConstraintSet,
```

Comment:
Should this be a list? On lines 461-463 we have

```python
            constraints = eq_constraints
            if current_constraints:
                constraints += current_constraints
```
and I don't think `MagneticConstraintSet` implements `+=`.

Also, `constraints = eq_constraints` doesn't create a copy, should it?

I think we need to throw some sort of error if `self._eq_prob_cls` isn't either of the two types checked in the `if` statements, otherwise we're going to get a `NameError` because `problem` is not defined.

Reply:
It should be `List[MagneticConstraint]`, sure I'll add an error